nodeup/pkg/model/cloudconfig.go

```diff
@@ -38,6 +44,31 @@ func (b *CloudConfigBuilder) Build(c *fi.ModelBuilderContext) error {
 	var lines []string
 
 	cloudConfig := b.Cluster.Spec.CloudConfig
+
+	if fi.CloudProviderID(b.Cluster.Spec.CloudProvider) == fi.CloudProviderVSphere {
```

Comment:
I missed this point in earlier reviews, why not separate out cloud config creation based on cloud provider type?
Eg: 
if fi.CloudProviderID(b.Cluster.Spec.CloudProvider) == fi.CloudProviderVSphere {
...
} else if fi.CloudProviderID(b.Cluster.Spec.CloudProvider) == fi.CloudProviderGCE {
...
} else {
...
}

Current code is vSphere entries, then GCE, then again vSphere code.